File: Intro/CS50/pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void){
    float money;
    printf("O hai! How much change is owed?\n...$");
    do{
        money = GetFloat();
        if(money < 0){
            printf("Please enter a positive number.\n");
        };
    }while(money < 0);
    money = (int)round(money*100);
    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    while(money > 24){
        quarters++;
        money -= 25;
    };
    while(money > 9){
        dimes++;
        money -= 10;
    };
    while(money > 4){
        nickels++;
        money -= 5;
    };
    //Print for check50:
    int total_change = quarters + dimes + nickels + money;
    printf("%.0d\n", total_change);
}

File: Intro/CS50/pset6/greedy.py
# get change in (formatted)
print ("O hai! How much change is owed?")
while True:
    try:
        change = float(input())
    except ValueError:
        print ("How much change is owed?")
    else:
        if change > 0:
            break
        print("How much change is owed?")
        
change = int(change * 100)

# get quarters out
quarters = (change - (change % 25))/25
change = change - (quarters * 25)

# get dimes out
dimes = (change - (change % 10))/10
change = change - (dimes * 10)

# get nickels out
nickels = (change - (change % 5))/5
change = change - (nickels * 5)

# pennies out is whatever's left of change

# long format print
#print ("Quarters out : {}\nDimes out: {}\nNickels out: {}\nPennies out: {}"
#       .format(int(quarters), int(dimes), int(nickels), int(change)))

# short format print
print ("Total change out: {}".format(int(quarters + dimes + nickels + change)))
